src/rpc/estimation/gasEstimationHandler.ts

```diff
@@ -11,38 +11,34 @@ export const EXECUTE_SIMULATOR_BYTECODE =
 
 function getStateOverrides({
+    addSenderBalanceOverride,
     userOperation,
     entryPoint,
     replacedEntryPoint,
     stateOverride = {}
 }: {
+    addSenderBalanceOverride: boolean
     entryPoint: Address
     replacedEntryPoint: boolean
     stateOverride: StateOverrides
     userOperation: UserOperation
 }) {
-    return replacedEntryPoint
-        ? {
-              ...stateOverride,
-              [userOperation.sender]: {
-                  balance: toHex(100000_000000000000000000n),
-                  ...(stateOverride
-                      ? deepHexlify(stateOverride?.[userOperation.sender])
-                      : [])
-              },
-              [entryPoint]: {
-                  code: EXECUTE_SIMULATOR_BYTECODE
-              }
-          }
-        : {
-              ...stateOverride,
-              [userOperation.sender]: {
-                  balance: toHex(100000_000000000000000000n),
-                  ...(stateOverride
-                      ? deepHexlify(stateOverride?.[userOperation.sender])
-                      : [])
-              }
-          }
+    const result: StateOverrides = { ...stateOverride }
+
+    if (addSenderBalanceOverride) {
+        result[userOperation.sender] = {
+            balance: toHex(100000_000000000000000000n),
+            ...deepHexlify(stateOverride?.[userOperation.sender] || {})
```

Comment:
if user has passed balance `userOperation.sender` will this not override our balance override for first estimation?